assets/scripts/datalayer/common.js

```diff
@@ -103,6 +115,51 @@ document.shopAnalytics = {
       }
       window.dataLayer.push(event_data);
     }
+  },
+
+  cart: {
+    elements: {
+      product: '.shop-analytics-product-info',
+      shippingMethods: 'select.shipping_method, input[name^="shipping_method"]',
+      shippingMethodSelected: 'select.shipping_method:selected, input[name^="shipping_method"]:checked',
+      paymentMethods: 'select.payment_method, input[name^="payment_method"]',
+      paymentMethodSelected: 'select.payment_method:selected, input[name^="payment_method"]:checked'
+    }
+  },
+  checkout: {
+    dataInit: {
+      event: 'EECcheckout',
+      ecommerce: {
+        checkout: {
+          actionField: {
+            step: 0,
+          },
+          products: []
+        }
+      }
+    },
+    elements: {
+      shippingMethods: 'select.shipping_method, input[name^="shipping_method"]',
+      shippingMethodSelected: 'select.shipping_method:selected, input[name^="shipping_method"]:checked',
+      paymentMethods: 'input[name="payment_method"]',
+      paymentMethodSelected: 'select.payment_method:selected, input[name="payment_method"]:checked',
+      billingAddressFields: '.woocommerce-billing-fields input.input-text, .woocommerce-billing-fields select',
+      billingAddressFieldsRequired: '.woocommerce-billing-fields .validate-required input.input-text, .woocommerce-billing-fields .validate-required select',
+      shippingAddressFields: '.woocommerce-shipping-fields input.input-text, .woocommerce-shipping-fields select',
+      shippingAddressFieldsRequired: '.woocommerce-shipping-fields .validate-required input.input-text, .woocommerce-shipping-fields .validate-required select',
+      shippingAddressToggle: '#ship-to-different-address-checkbox'
+    },
+    steps: {
+      viewCart: 1,
+      shippingMethod: 2,
+      billingAddress: 3,
+      paymentMethod: 4,
+      shippingAddress: 5 
```

Comment:
The standard order is cart, account, billing, shipping, payment, confirmation.

Reply:
@sun I just followed the guidelines in https://docs.google.com/document/d/1Wpe4YoVDrdgiXtzfuSmIh36rEpaJ0gM87LMKST8BacQ/edit#. I will try to fix this if that is wrong, though.

Reply:
Unless I misunderstand the concept, the idea is that the steps are always in the order in which they actually appear in the front-end, and the "step" number in "actionField" passed in the event automatically follows that order, too.

This is the most important information of this entire tracking; i.e., at which exact point did the user abort the checkout process.

After studying the code again now, it seems like the step numbers are currently hardcoded in the order declared here, so it would not be possible to extract sensible reports out of this (unless the steps happen to be exactly in this order on one of the customer websites; e.g. GACO).

Reply:
@sun I think I get what you mean and I understand about the importance of recording the actual step where the event has been triggered. The problem I see here is that those steps can vary substantially from site to site (exist or nor, modified order, multiple steps mixed in one...) and I really don't see how to manage this.

Can I ask you for a short hangouts chat next week to discuss this? Perhaps even with Stephan, as he  may be able to clarify (at least to me) some concepts and requirements.